### PR TITLE
TOFCMOS 589

### PR DESCRIPTION
**Pull request by rbudai98**
PR:
* Adding optimization flag for compilation (-Ofast for all platforms and -march=armv8-a+simd for NXP)
* Copy frame buffer before passing on to toficompute to have faster access from user space

### CODE REVIEW

**Comment by rbudai98** [comment]
Measurements:

Without memcp: 
	Open source:
		ComputeXYZ took:3486us
		DeInterleaveDepth took:142886us
		I20240328 10:35:59.481987 205613 main.cpp:450] Measured FPS: 6.53172
	Closed source:
		ComputeXYZ took:3461us
		I20240328 11:09:55.011551 213023 main.cpp:494] Measured FPS: 16.1138
With memcpy:
	Open source:
		ComputeXYZ took:3699us
		DeInterleaveDepth took:4363us
		I20240328 11:17:02.536005 213839 main.cpp:494] Measured FPS: 19.0588

	CLosed source:
		ComputeXYZ took:4276us
		I20240328 11:14:25.893596 213343 main.cpp:494] Measured FPS: 19.1471

File: sdk/src/connections/target/buffer_processor.cpp [hunk 1]
```diff
@@ -211,6 +211,7 @@ aditof::Status BufferProcessor::processBuffer(uint16_t *buffer = nullptr) {
     unsigned int buf_data_len;
     uint8_t *pdata;
     dev = m_inputVideoDev;
+    uint8_t *pdata_user_space;
```
Comment (SeptimiuVana) [inline sdk/src/connections/target/buffer_processor.cpp:214]: why is this needed?

Comment (rbudai98) [reply sdk/src/connections/target/buffer_processor.cpp:214]: In case we copy the values on pdata in a new pointer, and than pass it on to tofiCompute, we can have a much faster processing time.

**Comment by dNechita** [comment]
@rbudai98 can you do one more check please? It is useful to know the impact for the uses case where the frames are MP. Thanks!

**Comment by rbudai98** [comment]
MP measurements:
Without memcp: 
>	Open source:
>>		ComputeXYZ took:15708us
>>		DeInterleaveDepth took:578779us
>>		I20240328 14:05:10.772272 227683 main.cpp:494] Measured FPS: 1.64547
>	Closed source:
>>		ComputeXYZ took:14931usComputeXYZ()
>>		I20240328 14:02:49.846947 227296 main.cpp:494] Measured FPS: 9.54499

With memcpy:
>	Open source:
>>		DeInterleaveDepth took:16678us
>>		I20240328 13:57:09.718338 226313 main.cpp:494] Measured FPS: 10.0465
>	CLosed source:
>>		ComputeXYZ took:15052usComputeXYZ()
>>		I20240328 13:59:50.923023 226753 main.cpp:494] Measured FPS: 10.2909

**Comment by dNechita** [comment]
> MP measurements: Without memcp:
> 
> > Open source:
> > > ```
> > >  ComputeXYZ took:15708us
> > >  DeInterleaveDepth took:578779us
> > >  I20240328 14:05:10.772272 227683 main.cpp:494] Measured FPS: 1.64547
> > > ```
> > 
> > 
> > Closed source:
> > > ```
> > >  ComputeXYZ took:14931usComputeXYZ()
> > >  I20240328 14:02:49.846947 227296 main.cpp:494] Measured FPS: 9.54499
> > > ```
> 
> With memcpy:
> 
> > Open source:
> > > ```
> > >  DeInterleaveDepth took:16678us
> > >  I20240328 13:57:09.718338 226313 main.cpp:494] Measured FPS: 10.0465
> > > ```
> > 
> > 
> > CLosed source:
> > > ```
> > >  ComputeXYZ took:15052usComputeXYZ()
> > >  I20240328 13:59:50.923023 226753 main.cpp:494] Measured FPS: 10.2909
> > > ```

Thank you! I appreciate it.

File: sdk/src/connections/target/buffer_processor.cpp (hunk 1 above)
Comment (dNechita) [reply sdk/src/connections/target/buffer_processor.cpp:214]: I would initialize this to nullptr.

**Comment by rbudai98** [comment]
v2:
* pdata_user_space initialized to nullptr

**Review by liealex — APPROVED** [review]

**Review by dNechita — APPROVED** [review]
